src/strategies/slack.js

```diff
@@ -0,0 +1,266 @@
+ * @property {SlackMessage} message The message data.
+ */
+
+/**
+ * @typedef {Object} SlackMessage
+ * @property {string} text The text of the message.
+ * @property {string} user The user ID who posted the message.
+ * @property {string} ts The timestamp of the message.
+ */
+
+/**
+ * @typedef {Object} SlackErrorResponse
+ * @property {boolean} ok Whether the request was successful (false for errors).
+ * @property {string} error The error code.
+ */
+
+/**
+ * @typedef {Object} SlackUploadResponse
+ * @property {boolean} ok Whether the request was successful.
+ * @property {SlackFile} file The uploaded file data.
+ */
+
+/**
+ * @typedef {Object} SlackFile
+ * @property {string} id The file ID.
+ * @property {string} name The file name.
+ * @property {string} permalink The permanent link to the file.
+ */
+
+//-----------------------------------------------------------------------------
+// Constants
+//-----------------------------------------------------------------------------
+
+const API_BASE = "https://slack.com/api";
+
+//-----------------------------------------------------------------------------
+// Exports
+//-----------------------------------------------------------------------------
+
+/**
+ * A strategy for posting messages to Slack.
+ */
+export class SlackStrategy {
+	/**
+	 * Maximum length of a Slack message in characters.
+	 * @type {number}
+	 * @const
+	 */
+	MAX_MESSAGE_LENGTH = 4000;
+
+	/**
+	 * The ID of the strategy.
+	 * @type {string}
+	 * @readonly
+	 */
+	id = "slack";
+
+	/**
+	 * The display name of the strategy.
+	 * @type {string}
+	 * @readonly
+	 */
+	name = "Slack";
+
+	/**
+	 * The options for this strategy.
+	 * @type {SlackOptions}
+	 */
+	#options;
+
+	/**
+	 * Creates a new instance.
+	 * @param {SlackOptions} options Options for the instance.
+	 * @throws {Error} When options are missing.
+	 */
+	constructor(options) {
+		const { botToken, channel } = options;
+
+		if (!botToken) {
+			throw new TypeError("Missing bot token.");
+		}
+
+		if (!channel) {
+			throw new TypeError("Missing channel.");
+		}
+
+		this.#options = options;
+	}
+
+	/**
+	 * Calculates the length of a message according to Slack's algorithm.
+	 * All characters are counted as is.
+	 * @param {string} message The message to calculate the length of.
+	 * @returns {number} The calculated length of the message.
+	 */
+	calculateMessageLength(message) {
+		return [...message].length;
+	}
+
+	/**
+	 * Uploads an image to Slack.
+	 * @param {Uint8Array} imageData The image data to upload.
+	 * @param {string} filename The filename for the image.
+	 * @param {string} [altText] The alt text for the image.
+	 * @param {PostOptions} [postOptions] Additional options for the upload.
+	 * @returns {Promise<SlackUploadResponse>} A promise that resolves with the upload response.
+	 * @throws {Error} When the upload fails.
+	 */
+	async #uploadImage(imageData, filename, altText, postOptions) {
+		const url = `${API_BASE}/files.upload`;
```

Comment:
This API no longer works. See https://docs.slack.dev/messaging/working-with-files#uploading_files for how to do this correctly.